src/feature/relay/router.c

```diff
@@ -829,30 +829,36 @@ router_initialize_tls_context(void)
  * -1 if Tor should die,
  */
 STATIC int
-router_write_fingerprint(int hashed)
+router_write_fingerprint(int hashed, int ed25519_identity)
 {
   char *keydir = NULL, *cp = NULL;
   const char *fname = hashed ? "hashed-fingerprint" :
-                               "fingerprint";
+                      (ed25519_identity ? "ed25519_identity" : "fingerprint");
```

Comment:
Not sure you can do that. What happens if both `hashed = 1` and `ed25519_identity = 1 ` ?

I think you need to keep the `hashed` as it is and then individually test `ed25519_identity` after so you can write more than one file.